Asthma/TabBarControllers/Dashboard/APHCalendarDataModel.m

```diff
@@ -194,128 +228,85 @@ -(NSDictionary *)userCompliedWithDailyScheduledTasks:(APHCalendarTaskType)task i
         
         dateRange = [[APCDateRange alloc]initWithStartDate:startDate endDate:endDate];
         
-        //get array of completed scheduled tasks
-        NSArray *completedScheduledTasks = [APHCalendarDataModel allScheduledDailyPromptTasksForDateRange:dateRange completed:@YES inContext:((APCAppDelegate *)[UIApplication sharedApplication].delegate).dataSubstrate.mainContext];
-        
-        for (APCScheduledTask *task in completedScheduledTasks) {
-            NSDate *startOn = task.startOn;
+        NSMutableDictionary *complianceDictionary = [[NSMutableDictionary alloc]init];
+        NSArray *scheduledTasks = [self dailySurveyScheduledTasksForDateRange:dateRange];
+        for(APCScheduledTask *scheduledTask in scheduledTasks){
             
+            NSDate *startOn = scheduledTask.startOn;
             if ([startOn earlierDate:[NSDate new]] == startOn) {//no need to parse scheduled tasks after today
-                NSString * resultSummary = task.lastResult.resultSummary;
+                NSString * resultSummary = scheduledTask.lastResult.resultSummary;
                 NSDictionary * dictionary = resultSummary ? [NSDictionary dictionaryWithJSONString:resultSummary] : nil;
                 NSDateComponents *weekdayComponents = [gregorian components:(NSCalendarUnitDay) fromDate:startOn];
                 int day = (int)[weekdayComponents day];
-                if (dictionary[kDaytimeSickKey]) {//had symptoms, red == 0
+                if ([dictionary[kDaytimeSickKey] isEqualToNumber: @1]) {//had symptoms
                     [complianceDictionary setObject:@"0" forKey:[NSString stringWithFormat:@"%i", day]];
-                }else{
-                    [complianceDictionary setObject:@"1" forKey:[NSString stringWithFormat:@"%i", day]];
+                }else if ([dictionary[kDaytimeSickKey] isEqualToNumber: @0]){
+                    [complianceDictionary setObject:@"1" forKey:[NSString stringWithFormat:@"%i", day]];//1 = compliance/green color
                 }
             }
         }
+        
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [[NSNotificationCenter defaultCenter]postNotificationName:calendarDataSourceDidUpdateComplianceDictionaryNotification object:complianceDictionary];
+        });
     }
-    return complianceDictionary;
 }
 
-+ (NSArray *)allScheduledTasksForDateRange: (APCDateRange*) dateRange completed: (NSNumber*) completed inContext: (NSManagedObjectContext*) context
-{
-    NSFetchRequest * request = [APCScheduledTask request];
-    request.shouldRefreshRefetchedObjects = YES;
-    NSPredicate * datePredicate = [NSPredicate predicateWithFormat:@"(startOn >= %@) AND (endOn <= %@)", dateRange.startDate, dateRange.endDate];
+#pragma mark Helpers
+-(BOOL)isValidDateForDay:(NSInteger)DD month:(NSInteger)MM year:(NSInteger)YYYY{
```

Comment:
Make this a shared utility method, I have seen this method twice.

![screen shot 2015-06-22 at 1 48 40 pm](https://cloud.githubusercontent.com/assets/11466704/8292601/6cbca006-18e5-11e5-9fef-531618bd4bbd.png)